.github/workflows/build.yaml

```diff
@@ -190,10 +193,13 @@ jobs:
       package-name: cuopt_sh_client
       package-type: python
       append-cuda-suffix: false
+      pure-wheel: true
+      # only need 1 build (noarch package): this selects amd64, oldest-supported Python, latest-supported CUDA
+      matrix_filter: '[map(select(.ARCH == "amd64")) | min_by((.PY_VER | split(".") | map(tonumber)), (.CUDA_VER | split(".") | map(-tonumber)))]'
```

Comment:
*(moved to a corrected version of this filter)*

Changes like this fix the errors about conflicting artifact names. For pure-Python packages, we only need to build 1 time... not once per combination of Python version, CUDA versions, operating system, and CPU architecture.

This change should save `cuopt` some CI time by avoiding lots of unnecessary wheel builds 😁